### ISSUE TITLE
Make Team->Survivor relationship many-to-many

### ISSUE THREAD

**Issue by mawilson** [mawilson/survivor_draft#105]
Currently there's a problem with the DB schema where the Survivors had to be duplicated to account for season 46a & 46b. This was a nuisance with two teams, & can potentially cause problems with the data between the two differing when they should be identical. However, it will really be a problem when/if even more subseasons per season get added, as then we're talking about a copy of each survivor for each subseason.

I still think that having separate entries for each subseason is okay, although it may make sense to link them in some way at some point (such as, for instance, grouping subseasons together on the homepage so as to show all drafted teams for a single season on the same page). However, duplicating the survivors is clearly suboptimal.

I think the simple solution is to turn the Survivor 'Team' field from a ForeignKey to a ManyToMany. Logically this makes sense. Now that subseasons exist, a Team can have multiple Survivors (as before), & a Survivor can have multiple Teams (say, one in Season 46a, one in Season 46b).

One possible complication to this is the Fan Favorite vote, & is something I'll need to look into. I'm not sure if this will mean all votes will go towards a team's subseason, or if subseasons will effectively add together - & I think both are valid for the purposes of a fan favorite vote. Maybe it could end up being a Rubric Boolean?

In short:
* Make Survivor -> Team relationship ManyToMany
* Investigate impact on Fan Favorite vote
* Maybe add Rubric option for collective fan favorite vote or specific to the subseason vote

**Comment by mawilson** [comment]
Implemented in https://github.com/mawilson/survivor_draft/pull/109, minus fan favorite considerations

**Comment by mawilson** [comment]
Fan favorite Share Vote Rubric Boolean added in https://github.com/mawilson/survivor_draft/pull/139, now done
